src/client/opamCommands.ml

```diff
@@ -1520,6 +1517,14 @@ let config cli =
                | Some v -> Printf.sprintf " (default repo at %s)" v
                | None -> ""
               );
+          in
+          OpamRepositoryState.with_ `Lock_none gt @@ fun rt ->
+          print_repositories "all-repositories" rt;
+          match OpamStateConfig.get_switch_opt () with
+          | None -> print "current-switch" "%s" "none set"; `Ok ()
+          | Some switch ->
+            OpamSwitchState.with_ `Lock_none ~switch gt @@ fun state ->
+            print_repositories "repositories" state.switch_repos;
```

Comment:
I don't understand the difference between `all-repositories` and `repositories` fields

Reply:
`repositories` only shows the list of repositories from the current switch.
`all-repositories` shows all of them (as in: `opam repository list -a`)

Reply:
Hmm... It is then the wrong value used: `rt` = `state.switch_repos`, to access to switch state repos, it is `state.switch_config.repos`